internal/db/update/schema.go

```diff
@@ -190,37 +214,50 @@ ORDER BY name
 }
 
 // Apply any pending update that was not yet applied.
-func ensureUpdatesAreApplied(ctx context.Context, tx *sql.Tx, current int, updates []schema.Update, hook schema.Hook) error {
-	if current > len(updates) {
-		return fmt.Errorf(
-			"schema version '%d' is more recent than expected '%d'",
-			current, len(updates))
+func ensureUpdatesAreApplied(ctx context.Context, tx *sql.Tx, versions []int, schemaUpdates map[updateType][]schema.Update, hook schema.Hook) error {
+	if versions == nil {
+		versions = []int{0, 0}
 	}
 
-	// If there are no updates, there's nothing to do.
-	if len(updates) == 0 {
-		return nil
-	}
+	for updateType, updates := range schemaUpdates {
```

Comment:
Do we need to iterate schemaUpdates such that internal updates are processed first, rather than in random order?